src/main/kotlin/com/toasttab/pulseman/pulsar/Pulsar.kt

```diff
@@ -133,38 +140,45 @@ class Pulsar(
             try {
                 return mapper.readValue(propertiesJsonMap, mapTypeRef)
             } catch (ex: Exception) {
-                setUserFeedback("Failed to deserialize properties=$propertiesJsonMap. Error=$ex")
+                setUserFeedback("$FAILED_TO_DESERIALIZE_PROPERTIES=$propertiesJsonMap. $EXCEPTION=$ex")
             }
         }
         return emptyMap()
     }
 
     fun sendMessage(message: ByteArray?) {
         if (pulsarSettings.serviceUrl.value.isBlank()) {
-            setUserFeedback("Service Url is not set")
+            setUserFeedback(SERVICE_URL_NOT_SET)
             return
         }
         val topic = pulsarSettings.topic.value
         if (topic.isBlank()) {
-            setUserFeedback("Topic is not set")
+            setUserFeedback(TOPIC_NOT_SET)
             return
         }
         if (message == null) {
-            setUserFeedback("No class generated to send")
+            setUserFeedback(NO_CLASS_GENERATED_TO_SEND)
             return
         }
 
-        return try {
-            val messageId = createNewProducer(topic)
+        try {
+            createNewProducer(topic)
                 ?.newMessage()
                 ?.value(message)
                 ?.eventTime(System.currentTimeMillis())
                 ?.properties(properties())
                 ?.send()
-
-            setUserFeedback("Message sent with id $messageId on topic $topic")
+                ?.let { messageId ->
+                    setUserFeedback("$MESSAGE_SENT_ID $messageId $ON_TOPIC $topic")
+                }
         } catch (ex: Throwable) {
-            setUserFeedback("Could not transmit on topic $topic.\nError:$ex")
+            setUserFeedback("$COULD_NOT_TRANSMIT_TOPIC $topic.\n$EXCEPTION:$ex")
         }
     }
+
+    companion object {
+        private val mapper = ObjectMapper().registerModule(KotlinModule.Builder().build())
```

Comment:
Would it be worth to have a single instance of `ObjectMapper` in the application instead of creating a new one every time?

Reply:
The three of them have slightly different settings so I was reluctant to do it, they could probably be configured into one if I spent the time to make sure the settings all worked perfect together in the three use cases, I'm going to leave this for now, but its a good point.